.github/workflows/daily-flock-rebase.yaml

```diff
@@ -37,128 +37,18 @@ jobs:
 
-      - name: Checkout Flock
-        uses: actions/checkout@v4
+      - name: Assemble Flock
+        uses: ./.github/actions/assemble-flock
         with:
-          path: 'nest/flock'
-          fetch-depth: 0 # Checkout everything to get access to the tags
-          repository: ${{ env.FLOCK_REPO }}
-          ref: ${{ env.FLUTTER_BRANCH }}
-          token: ${{ env.GITHUB_PAT }}
-
-      - name: Fetch Flutter branch
-        working-directory: nest/flock
-        run: |
-          git config --global user.email "${{ github.actor }}@users.noreply.github.com"
-          git config --global user.name "${{ github.actor }}"
-
-          # Setup upstream to Flutter and fetch the latest from the desired branch.
-          git remote add upstream https://github.com/flutter/flutter.git
-          git fetch upstream ${{ env.FLUTTER_BRANCH }}
-
-      - name: Lookup Flutter branch tag (beta, stable)
-        if: ${{ env.FLUTTER_BRANCH == 'beta' || env.FLUTTER_BRANCH == 'stable' }}
-        working-directory: nest/flock
-        run: |
-          commit=$(git ls-remote upstream refs/heads/${{ env.FLUTTER_BRANCH }} | awk '{print $1}')
-          tag=$(git ls-remote --tags upstream | grep "$commit" | awk -F'/' '{print $NF}')
-          
-          echo "${{ env.FLUTTER_BRANCH }} commit: $commit"
-          echo "${{ env.FLUTTER_BRANCH }} tag: $tag"
-          echo "tag=$tag" >> $GITHUB_ENV
-
-      - name: Delete Flock branch, recreate, replay Flock
-        working-directory: nest/flock
-        run: |
-          # Checkout a temporary branch so we can manipulate Flock.
-          git checkout -b temp
-
-          # Delete Flock branch
-          git branch -D ${{ env.FLUTTER_BRANCH }}
-
-          # Get fresh version from Flutter.
-          git switch -c ${{ env.FLUTTER_BRANCH }} upstream/${{ env.FLUTTER_BRANCH}}
+          flutter_ref: ${{ matrix.config }}
+          token: ${{ secrets.REPO_WORKFLOW_PAT }}
 
-          # Switch back to the refreshed desired branch.
-          git checkout ${{ env.FLUTTER_BRANCH }}
-
-      - name: Tag latest master commit
-        if: ${{ env.FLUTTER_BRANCH == 'master' }}
-        working-directory: nest/flock
-        run: |
-          # Normally, Flutter looks for an engine version with the same hash as the
-          # latest commit. But with Flock, the Flock commits are always the latest,
-          # and there is no corresponding engine build. To work around this, we tag
-          # Flutter's latest commit on `master`, and then we adjust the script at
-          # `bin/internal/update_dart_sdk.sh` to use the tagged hash for the engine.
-          echo "Tagging most recent Flutter commit (flock-engine-hash): $(git rev-parse HEAD)"
-
-          git tag -d flock-engine-hash
-          git tag -a flock-engine-hash -m "Engine hash for Flock"
-
-          git show flock-engine-hash
-
-      - name: Setup Python
-        uses: actions/setup-python@v5
+      - name: push build to branch
```

Comment:
When reading this step name, it's not clear to me what the impact is, or why we're doing it. Is there a more accurate name? Or one that better illustrates intent?